packages/serverpod/lib/src/server/endpoint_dispatch.dart

```diff
@@ -332,6 +333,7 @@ abstract class EndpointDispatch {
   ///
   /// Throws an exception if required parameters are missing or if the
   /// paramString can't be jsonDecoded.
+  @visibleForTesting
```

Comment:
Comment: I'm generally not a fan of this type of thing, I think you should always test the public interfaces only and as far up the call stack as possible within the unit. 

To me this indicates that we either have too coupled code (which I guess is very likely in this case), or that we are testing on the wrong level.

Is it possible to refactor to make this testable in a reasonable way?

Reply:
I do agree! I think this is one of those functions which you do really want to have a detailed unit test for. As Alex pointed out here, we could move this to a helper file that is private to the package but still public for testing: https://github.com/serverpod/serverpod/pull/2676#discussion_r1732996759.